File: accis_run.cpp
#include "accis.hpp"

void accis_run(std::vector<accis_sat>& sats, int steps) {

    int num_sat = sats.size();

    for (int k = 0; k < steps; k++) {

        for (int i = 0; i < num_sat; i++)
            sats[i].step();

        for (int i = 0; i < num_sat; i++)
            for (int j = 0; j < num_sat; j++)
                if (i != j)
                    sats[i].transmit(sats[j]);

    }

}

File: accis_run.py
import time

import build.accis as accis

from accis_io import *

def accis_run(sat_list, num_steps, save_cadence) :
    print("Running ACCIS")
    ti = time.time()
    for k in range(num_steps) :
        if (k % 10 == 0) :
            print('Step ' + str(k))
        for sat in sat_list :
            sat.step()
        for sat1 in sat_list :
            for sat2 in sat_list :
                sat1.transmit(sat2)
        if (k != 0 and k % save_cadence == 0) :
            accis_save_results(sat_list)
    tf = time.time()
    run_time = round((tf - ti) / 60, 1)
    print("Run Time (min): %s" % run_time)
